### DISCUSSION TITLE
SAGPool implementation and Documentation difference

### DISCUSSION THREAD

**Discussion by arake001** [pyg-team/pytorch_geometric#3916] [Q&A] [answered]
Hello!

I have a question regarding the implementation of sagpool and the documentation provided for the sagpool.

`attn = x if attn is None else attn
attn = attn.unsqueeze(-1) if attn.dim() == 1 else attn
score = self.gnn(attn, edge_index).view(-1)

if self.min_score is None:
    score = self.nonlinearity(score) 
else:
    score = softmax(score, batch)

perm = topk(score, self.ratio, batch, self.min_score)`

Here, the **score values** given to the **topk** pooling are after applying the non-linearity function. But, the documentation suggests the score before applying the non-linearity function. 
 
**Documentation:**

**y = GNN(X, A)
I = topk(y)
X' = (X dot product   tanh(y))**

Can anyone explain this?

Thank you in advance for your help.

**Comment by wsad1** [comment, marked as answer]
topk(y) and topk(tanh(y)) will return the same set of nodes, in other words applying tanh will not change the ranking. Because tanh is a increasing function i.e is if if y1>y2 then tanh(y1)>tanh(y2).
